shared/src/business/useCases/courtIssuedDocument/fileAndServeCourtIssuedDocumentInteractor.test.ts

```diff
@@ -184,35 +186,135 @@ describe('fileAndServeCourtIssuedDocumentInteractor', () => {
   });
 
   it('should create a deadline on the subject case when docket entry is an Order For Filing Fee', async () => {
+    const mockOrderForFilingFee = {
+      ...mockDocketEntryWithWorkItem,
+      docketEntryId: 'ed68e80a-cab3-44b9-9280-03816bba8cba',
+      eventCode: orderFilingFeeEventCode,
+    };
+
     const mockOrderFilingFeeForm = {
       date: '2030-01-20T00:00:00.000Z',
       documentType: 'Order for Filing Fee',
-      eventCode: 'OF',
+      eventCode: orderFilingFeeEventCode,
     };
 
+    caseRecord.docketEntries.push(mockOrderForFilingFee);
+
     await fileAndServeCourtIssuedDocumentInteractor(applicationContext, {
-      clientConnectionId: 'testing',
-      docketEntryId: caseRecord.docketEntries[0].docketEntryId,
+      clientConnectionId: mockClientConnectionId,
+      docketEntryId: mockOrderForFilingFee.docketEntryId,
       docketNumbers: [caseRecord.docketNumber],
       form: mockOrderFilingFeeForm,
       subjectCaseDocketNumber: caseRecord.docketNumber,
     });
 
     expect(
-      applicationContext.getPersistenceGateway().createCaseDeadline.mock
-        .calls[0][0].caseDeadline,
+      applicationContext.getUseCaseHelpers().autoGenerateDeadline.mock
+        .calls[0][0],
     ).toMatchObject({
-      associatedJudge: caseRecord.associatedJudge,
       deadlineDate: mockOrderFilingFeeForm.date,
-      description: FILING_FEE_DEADLINE_DESCRIPTION,
-      docketNumber: caseRecord.docketNumber,
-      sortableDocketNumber: 18000101,
+      description:
+        SYSTEM_GENERATED_DOCUMENT_TYPES.orderForFilingFee.deadlineDescription,
     });
   });
 
   it('should NOT create a deadline on the subject case when docket entry is NOT an Order For Filing Fee', async () => {
     await fileAndServeCourtIssuedDocumentInteractor(applicationContext, {
-      clientConnectionId: 'testing',
+      clientConnectionId: mockClientConnectionId,
+      docketEntryId: caseRecord.docketEntries[0].docketEntryId,
+      docketNumbers: [caseRecord.docketNumber],
+      form: {
+        documentType: 'Order',
+        eventCode: 'O',
+      },
+      subjectCaseDocketNumber: caseRecord.docketNumber,
+    });
+
+    expect(
+      applicationContext.getUseCaseHelpers().autoGenerateDeadline,
+    ).not.toHaveBeenCalled();
+  });
+
+  it('should create a deadline on the subject case when docket entry is an Order For Amended Petition', async () => {
```

Comment:
NIt: Could these tests that check for OAP/OF generating a deadline be combined into one? My thinking is since there is an entity method checking the event code against a list of pre-defined codes that should generate a deadline, one test to make sure that method is called is sufficient